pkg/controller/controller_instance.go

```diff
@@ -89,20 +93,54 @@ const (
 	startingInstanceOrphanMitigationMessage string = "The instance provision call failed with an ambiguous error; attempting to deprovision the instance in order to mitigate an orphaned resource"
 
 	clusterIdentifierKey string = "clusterid"
+
+	minBrokerOperationRetryDelay time.Duration = time.Second * 1
+	maxBrokerOperationRetryDelay time.Duration = time.Minute * 20
 )
 
+type retryQueue struct {
+	mutex sync.RWMutex // lock to be used for accessing all retryQueue fields
+
+	// All keys in retryTime, backoffBeforeRetrying, and rateLimiter are in the form of
+	// namespace/instance@generation
+
+	retryTime map[string]time.Time // If instance is in map, this is the earliest
+	// time at which an operation re-attempt should be made to ensure we don't
+	// overwhelm the broker with repeated requests.
+
+	backoffBeforeRetrying map[string]time.Time // An instance is added immediately
```

Comment:
It would make things a lot cleaner if `retryTime` and `backoffBeforeRetrying` were combined into a single map whose value was a complex type. Then there is not extra wrangler to coordinate between the two and purge both maps.